src/collectors/snmpinterfacepoll/snmpinterfacepoll.py

```diff
@@ -73,11 +73,15 @@ def publish_metrics(self, gaugesList, countersList, timestamp):
 
         for gaugeDesc in gaugesList:
             metricPath, metricValue = gaugeDesc
-            metric = Metric(
-                            metricPath, metricValue,
-                            raw_value=None, timestamp=timestamp
-                            )
-            self.publish_metric(metric)
+            # Fixes devices that return no value
+            try:
+                metric = Metric(
```

Comment:
Creating a class probably doesn't need to be inside the 'try' block unless it has some propensity to fail or something.